test/stdlib/UnsafeRawBufferPointer.swift

```diff
@@ -0,0 +1,380 @@
+// RUN: %target-run-simple-swiftgyb
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var UnsafeRawBufferPointerTestSuite = TestSuite("UnsafeRawBufferPointer")
+
+// View an in-memory value's bytes.
+// Use copyBytes to overwrite the value's bytes.
+UnsafeRawBufferPointerTestSuite.test("initFromValue") {
+  var value1: Int32 = -1
+  var value2: Int32 = 0
+  // Immutable view of value1's bytes.
+  withUnsafeBytes(of: &value1) { bytes1 in
+    expectEqual(bytes1.count, 4)
+    for b in bytes1 {
+      expectEqual(b, 0xFF)
+    }
+    // Mutable view of value2's bytes.
+    withUnsafeMutableBytes(of: &value2) { bytes2 in
+      expectEqual(bytes1.count, bytes2.count)
+      bytes2[0..<bytes2.count].copyBytes(from: bytes1)
+    }
+  }
+  expectEqual(value2, value1)
+}
+
+// View an array's elements as bytes.
+// Use copyBytes to overwrite the array element's bytes.
+UnsafeRawBufferPointerTestSuite.test("initFromArray") {
+  var array1: [Int32] = [0, 1, 2, 3]
+  var array2 = [Int32](repeating: 0, count: 4)
+  // Immutable view of array1's bytes.
+  array1.withUnsafeBytes { bytes1 in
+    expectEqual(bytes1.count, 16)
+    for (i, b) in bytes1.enumerated() {
+      if i % 4 == 0 {
+        expectEqual(Int(b), i / 4)
+      }
+      else {
+        expectEqual(b, 0)
+      }
+    }
+    // Mutable view of array2's bytes.
+    array2.withUnsafeMutableBytes { bytes2 in
```

Comment:
The implementation of this tricky and it doesn't look like you're testing the tricky bits: ensuring uniqueness, and preventing undefined behavior if the user violates inout rules.